src/ipvs/ip_vs_service.c

```diff
@@ -113,10 +113,11 @@ struct dp_vs_service *__dp_vs_service_get(int af, uint16_t protocol,
     list_for_each_entry(svc, &dp_vs_svc_table[hash], s_list){
         if ((svc->af == af)
             && inet_addr_equal(af, &svc->addr, vaddr)
-            && (svc->port == vport)
             && (svc->proto == protocol)) {
-                rte_atomic32_inc(&svc->usecnt);
-                return svc;
+                if ((svc->port == vport) || (svc->port == 0)) {
+                    rte_atomic32_inc(&svc->usecnt);
+                    return svc;
+                }
```

Comment:
function dp_vs_set_svc use __dp_vs_service_get. If dpvs has service 192.168.1.1:0 and we use command ipvasdm -ln -t 192.168.1.1:80, ipvsadm will show service wrong